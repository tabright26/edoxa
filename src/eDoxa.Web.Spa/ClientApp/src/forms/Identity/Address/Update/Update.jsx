import React from "react";
import { Field, reduxForm } from "redux-form";
import { FormGroup, Col, Input, Form } from "reactstrap";

import Button from "../../../../components/Button";
import { DELETE_ADDRESS_FORM } from "../forms";

const UpdateAddressForm = ({ handleSubmit, handleCancel }) => (
  <Form onSubmit={handleSubmit}>
    <FormGroup>
      <Field type="text" name="country" component={({ input }) => <Input {...input} bsSize="sm" disabled />} />
    </FormGroup>
    <FormGroup>
      <Field type="text" name="line1" component={({ input }) => <Input {...input} placeholder="Address line 1" bsSize="sm" />} />
    </FormGroup>
    <FormGroup>
      <Field type="text" name="line2" component={({ input }) => <Input {...input} placeholder="Address line 2 (optional)" bsSize="sm" />} />
    </FormGroup>
    <FormGroup>
      <Field type="text" name="city" component={({ input }) => <Input {...input} placeholder="City" bsSize="sm" />} />
    </FormGroup>
    <FormGroup row className="my-0">
      <Col xs="8">
        <FormGroup>
          <Field type="text" name="state" component={({ input }) => <Input {...input} placeholder="State" bsSize="sm" />} />
        </FormGroup>
      </Col>
      <Col xs="4">
        <FormGroup>
          <Field type="text" name="postalCode" component={({ input }) => <Input {...input} placeholder="Postal Code" bsSize="sm" />} />
        </FormGroup>
      </Col>
    </FormGroup>
    <FormGroup className="mb-0">
      <Button.Save className="mr-2" />
      <Button.Cancel onClick={handleCancel} />
    </FormGroup>
  </Form>
);

export default reduxForm({ form: DELETE_ADDRESS_FORM })(UpdateAddressForm);
