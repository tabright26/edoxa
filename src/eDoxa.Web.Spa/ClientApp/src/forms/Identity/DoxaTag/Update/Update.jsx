import React from "react";
import { Field, reduxForm } from "redux-form";
import { FormGroup, Input, Form } from "reactstrap";
import Button from "../../../../components/Button";
import { UPDATE_DOXATAG_FORM } from "../forms";

const UpdateDoxaTagForm = ({ handleSubmit, handleCancel }) => (
  <Form onSubmit={handleSubmit}>
    <FormGroup>
      <Field type="text" name="name" component={({ input }) => <Input {...input} placeholder="DoxaTag" bsSize="sm" />} />
    </FormGroup>
    <FormGroup className="mb-0">
      <Button.Save className="mr-2" />
      <Button.Cancel onClick={handleCancel} />
    </FormGroup>
  </Form>
);

export default reduxForm({ form: UPDATE_DOXATAG_FORM })(UpdateDoxaTagForm);
