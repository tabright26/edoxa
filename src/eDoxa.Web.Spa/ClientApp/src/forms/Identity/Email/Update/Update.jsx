import React from "react";
import { Field, reduxForm } from "redux-form";
import { FormGroup, Input, Form } from "reactstrap";
import Button from "../../../../components/Buttons";
import { UPDATE_EMAIL_FORM } from "../forms";

const UpdateEmailForm = ({ handleSubmit, handleCancel }) => (
  <Form onSubmit={handleSubmit}>
    <FormGroup>
      <Field type="text" name="email" component={({ input }) => <Input {...input} placeholder="Email" bsSize="sm" />} />
    </FormGroup>
    <FormGroup className="mb-0">
      <Button.Save className="mr-2" />
      <Button.Cancel onClick={handleCancel} />
    </FormGroup>
  </Form>
);

export default reduxForm({ form: UPDATE_EMAIL_FORM })(UpdateEmailForm);
