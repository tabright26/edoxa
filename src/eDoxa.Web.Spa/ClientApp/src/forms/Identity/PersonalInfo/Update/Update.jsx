import React from "react";
import { reduxForm } from "redux-form";
import { FormGroup, Col, Input, Form } from "reactstrap";
import Button from "../../../../components/Button";
import { UPDATE_PERSONALINFO_FORM } from "../forms";

const UpdatePersonalInfoForm = ({ handleSubmit, handleCancel }) => (
  <Form onSubmit={handleSubmit}>
    <FormGroup row className="my-0">
      <Col xs="6">
        <FormGroup>
          <Input type="text" id="firstName" bsSize="sm" placeholder="Enter your first name" />
        </FormGroup>
      </Col>
      <Col xs="6">
        <FormGroup>
          <Input type="text" id="lastName" bsSize="sm" placeholder="Enter your last name" />
        </FormGroup>
      </Col>
    </FormGroup>
    <FormGroup>
      <Input className="d-inline" type="select" name="months" id="month" bsSize="sm" style={{ width: "60px" }}>
        <option value="0">mm</option>
        <option value="Male">Male</option>
        <option value="Female">Female</option>
        <option value="Other">Other</option>
      </Input>
      <span className="d-inline">/</span>
      <Input className="d-inline" type="select" name="days" id="day" bsSize="sm" style={{ width: "60px" }}>
        <option value="0">dd</option>
        <option value="Male">Male</option>
        <option value="Female">Female</option>
        <option value="Other">Other</option>
      </Input>
      <span className="d-inline">/</span>
      <Input className="d-inline" type="select" name="years" id="year" bsSize="sm" style={{ width: "75px" }}>
        <option value="0">YYYY</option>
        <option value="Male">Male</option>
        <option value="Female">Female</option>
        <option value="Other">Other</option>
      </Input>
    </FormGroup>
    <FormGroup>
      <Input type="select" name="gender" id="gender" bsSize="sm">
        <option value="0">Please select</option>
        <option value="Male">Male</option>
        <option value="Female">Female</option>
        <option value="Other">Other</option>
      </Input>
    </FormGroup>
    <FormGroup className="mb-0">
      <Button.Save className="mr-2" />
      <Button.Cancel onClick={handleCancel} />
    </FormGroup>
  </Form>
);

export default reduxForm({ form: UPDATE_PERSONALINFO_FORM })(UpdatePersonalInfoForm);
