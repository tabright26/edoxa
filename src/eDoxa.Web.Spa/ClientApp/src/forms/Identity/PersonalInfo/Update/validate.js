const validate = values => {
  const errors = {};
  if (!values.firstName) {
    errors.firstName = "First name is required.";
  }
  return errors;
};

export default validate;
