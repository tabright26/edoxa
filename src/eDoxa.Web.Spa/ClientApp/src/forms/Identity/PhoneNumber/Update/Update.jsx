import React from "react";
import { Field, reduxForm } from "redux-form";
import { FormGroup, Input, Form } from "reactstrap";
import Button from "../../../../components/Button";
import { UPDATE_PHONENUMBER_FORM } from "../forms";

const UpdatePhoneNumberForm = ({ handleSubmit, handleCancel }) => (
  <Form onSubmit={handleSubmit}>
    <FormGroup>
      <Field type="text" name="phoneNumber" component={({ input }) => <Input {...input} placeholder="Phone Number" bsSize="sm" />} />
    </FormGroup>
    <FormGroup className="mb-0">
      <Button.Save className="mr-2" />
      <Button.Cancel onClick={handleCancel} />
    </FormGroup>
  </Form>
);

export default reduxForm({ form: UPDATE_PHONENUMBER_FORM })(UpdatePhoneNumberForm);
