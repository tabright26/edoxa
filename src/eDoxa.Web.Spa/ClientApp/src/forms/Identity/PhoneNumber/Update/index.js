import UpdatePhoneNumberForm from "./Update";

export default UpdatePhoneNumberForm;
