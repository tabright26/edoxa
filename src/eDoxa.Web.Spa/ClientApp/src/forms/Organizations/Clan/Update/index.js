import UpdateAddressForm from "./Update";

export default UpdateAddressForm;
