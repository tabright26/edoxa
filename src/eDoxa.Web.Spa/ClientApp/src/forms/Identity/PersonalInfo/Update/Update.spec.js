import React from "react";
import Update from "./Update";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";

it("renders correctly", () => {
  const personalInfo = {
    firstName: "Boby",
    lastName: "Bob",
    birthDate: new Date(Date.now()),
    gender: "Male"
  };
  const tree = renderer
    .create(
      <Provider store={{ getState: () => {}, dispatch: action => {}, subscribe: () => {} }}>
        <Update personalInfo={personalInfo} />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
