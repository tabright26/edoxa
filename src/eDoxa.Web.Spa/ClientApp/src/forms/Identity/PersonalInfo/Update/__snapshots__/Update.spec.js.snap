// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <dl
    className="row mb-0"
  >
    <dd
      className="col-sm-3 text-muted mb-0"
    >
      Name
    </dd>
    <dd
      className="col-sm-9 mb-0"
    >
      <dl
        className="row"
      >
        <dt
          className="col-sm-6 mb-0"
        >
          <input
            className="form-control-sm form-control"
            name="firstName"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="Enter your first name"
            type="text"
            value=""
          />
          <div
            className="invalid-feedback"
          />
        </dt>
        <dd
          className="col-sm-6 mb-0"
        >
          <input
            className="form-control-sm form-control"
            disabled={true}
            type="text"
            value="Bob"
          />
        </dd>
      </dl>
    </dd>
    <dd
      className="col-sm-3 text-muted mb-0"
    >
      Birth date
    </dd>
    <dd
      className="col-sm-9 mb-0"
    >
      <div
        className="form-group"
      >
        <input
          className="d-inline form-control-sm form-control"
          disabled={true}
          name="day"
          style={
            Object {
              "width": "60px",
            }
          }
          type="text"
          value={1}
        />
        <span
          className="d-inline mx-2"
        >
          /
        </span>
        <input
          className="d-inline form-control-sm form-control"
          disabled={true}
          name="month"
          style={
            Object {
              "width": "60px",
            }
          }
          type="text"
          value={2}
        />
        <span
          className="d-inline mx-2"
        >
          /
        </span>
        <input
          className="d-inline form-control-sm form-control"
          disabled={true}
          name="year"
          style={
            Object {
              "width": "75px",
            }
          }
          type="text"
          value={51613}
        />
      </div>
    </dd>
    <dd
      className="col-sm-3 text-muted mb"
    >
      Gender
    </dd>
    <dd
      className="col-sm-5 mb-0"
    >
      <input
        className="form-control-sm form-control"
        disabled={true}
        name="Male"
        type="text"
        value="Male"
      />
      <button
        aria-label={null}
        className="mt-3 mr-2 btn btn-primary btn-sm"
        onClick={[Function]}
        type="submit"
      >
        Save
      </button>
      <button
        aria-label={null}
        className="mt-3 btn btn-outline-primary btn-sm"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </dd>
  </dl>
</form>
`;
